XCUITests/HomePageSettingsUITest.swift

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import XCTest
+
+let websiteUrl1 = "https://www.mozilla.org"
+let websiteUrl2 = "https://developer.mozilla.org/"
+let invalidUrl = "1-2-3"
+
+class HomePageSettingsUITests: BaseTestCase {
+    var navigator: Navigator!
+    var app: XCUIApplication!
+
+    override func setUp() {
+        super.setUp()
+        app = XCUIApplication()
+        navigator = createScreenGraph(app).navigator(self)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+    }
+
+    func testNavigation() {
+        // Check that the homepage menu exists
+        navigator.goto(HomePageSettings)
+        XCTAssertTrue(app.staticTexts["Homepage Settings"].exists)
+    }
+
+    private func enterWebPageAsHomepage(text: String) {
+        app.textFields["HomePageSettingTextField"].tap()
+        app.textFields["HomePageSettingTextField"].typeText(text)
+    }
+
+    func testTyping() {
+        navigator.goto(HomePageSettings)
+        // Enter a webpage
+        enterWebPageAsHomepage(text: websiteUrl1)
+
+        let value = app.textFields["HomePageSettingTextField"].value
+        XCTAssertEqual(value as? String, websiteUrl1, "The webpage typed does not match with the one saved")
```

Comment:
how about if we move this two lines inside enterWebPageAsHomepage() method? that way it would be better encapsulated.